pkg/fakerp/admin_handlers.go

```diff
@@ -1,58 +1,40 @@
 package fakerp
 
 import (
-	"encoding/json"
 	"net/http"
 
 	"github.com/go-chi/chi"
 
 	"github.com/openshift/openshift-azure/pkg/api"
 )
 
-func (s *Server) adminreply(w http.ResponseWriter, err error, out interface{}) {
-	if err != nil {
-		s.internalError(w, err.Error())
-		return
-	}
-
-	if out == nil {
-		return
-	}
-
-	if b, ok := out.([]byte); ok {
-		w.Header().Set("Content-Type", "application/octet-stream")
-		w.Write(b)
-		return
-	}
-
-	b, err := json.Marshal(out)
-	if err != nil {
-		s.internalError(w, err.Error())
-		return
-	}
-	w.Header().Set("Content-Type", "application/json")
-	w.Write(b)
-	return
-}
-
 // handleBackup handles admin requests to backup an etcd cluster
 func (s *Server) handleBackup(w http.ResponseWriter, req *http.Request) {
-	cs := s.read()
-	if cs == nil {
+	cs, err := s.store.Get()
+	if cs == nil || err != nil {
 		s.internalError(w, "Failed to read the internal config")
 		return
 	}
 
 	backupName := chi.URLParam(req, "backupName")
 
-	err := s.plugin.BackupEtcdCluster(req.Context(), cs, backupName)
+	err = s.store.UpdateState(api.Updating)
```

Comment:
why does the fake rp need to do this?  operations are serialised and non-crash-recoverable.  We will make our lives much easier if we don't attempt to do this.